### ISSUE TITLE
TypeScript definition for optional chaining misses undefined type

### ISSUE THREAD

**Issue by bennettdams** [microsoft/vscode#106947]
Issue Type: <b>Bug</b>

VS Code is not showing `undefined` for the type when looking at the definition of a variable that is filled via optional chaining with TypeScript.

Steps to reproduce:

1. Create a `test.ts` file.
2. Fill a variable via optional chaining.
3. Look at the definition (e. g. by hovering the variable)
=> It shows the type of value, but it should also show `undefined`.

Example:
```ts
interface Tesla {
    autopilot?: {
        isInstalled: boolean;
    };
}

const car: Tesla = {
    autopilot: undefined
};

const isAutopilotInstalled = car.autopilot?.isInstalled;

console.log(isAutopilotInstalled);
```

When hovering `isAutopilotInstalled` you'll see

`const isAutopilotInstalled: boolean`

but it should be

`const isAutopilotInstalled: boolean |  undefined`

as the optional chaining will return undefined under its conditions.

![image](https://user-images.githubusercontent.com/29319414/93497942-66a3ea00-f911-11ea-9f09-5e8209c56fef.png)

To see the right result you can use the [official TypeScript playground](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgCoQM4Bs7IN4CwAUMqcnAK5gD2ADsFtWAPwBc+xZXywGAkiAxg4WLBAAm7AEbVqYuCADcnMgF9lRVcWIJqgsMgRwo7dNlwBeDiTKUa9RmHYUQ4iDFATi67UV36eDABBKjoGJgEhETFxZCsjKAA6OzDHZkTeSOFRCQ0dPQw5CETGAHMACl4Q+3CwLOiJAEoNIA), where it correctly shows `boolean | undefined`:

![image](https://user-images.githubusercontent.com/29319414/93497998-791e2380-f911-11ea-8924-d9dd4469dfb0.png)

VS Code version: Code 1.49.0 (e790b931385d72cf5669fcefc51cdf65990efa5d, 2020-09-10T13:22:08.892Z)
OS version: Windows_NT x64 10.0.19041

**Comment by mjbvz** [comment]
Have you enabled strict null checks (or strict mode)?

**Comment by bennettdams** [comment]
> Have you enabled strict null checks (or strict mode)?

Strict null checks are not enabled.  Strict mode is enabled through modules by default I think.

**Comment by mjbvz** [comment]
Ah so that's the issue. Without [strict null checks](https://www.tektutorialshub.com/typescript/typescript-null-undefined-strict-null-checks/#strict-null-checks), TypeScript in effectively ignores `undefined` and `null`.

To fix this, just enable strict null checks (or [TypeScript's strict mode](https://medium.com/better-programming/how-typescripts-strict-mode-actually-fixes-typescript-736ba8108c85) which enables strict null checks)
